charts/turing/templates/_helpers.tpl

```diff
@@ -0,0 +1,248 @@
+{{- end -}}
+
+{{- define "turing.serviceAccount.name" -}}
+{{- include "turing.fullname" . -}}
+{{- end -}}
+
+{{- define "turing.mlp.encryption.key" -}}
+{{- .Values.config.MLPConfig.MLPEncryptionKey -}}
+{{- end -}}
+
+{{- define "turing.sentry.dsn" -}}
+{{- .Values.sentry.dsn -}}
+{{- end -}}
+
+{{- define "turing.plugins.directory" -}}
+/app/plugins
+{{- end -}}
+
+{{- define "turing.plugins.initContainers" -}}
+{{ if .Values.experimentEngines }}
+initContainers:
+{{ range $expEngine := .Values.experimentEngines }}
+{{ if eq (toString $expEngine.type) "rpc-plugin" }}
+- name: {{ $expEngine.name }}-plugin
+  image: {{ $expEngine.rpcPlugin.image }}
+  env:
+  - name: PLUGIN_NAME
+    value: "{{ $expEngine.name }}"
+  - name: PLUGINS_DIR
+    value: {{ include "turing.plugins.directory" . }}
+  volumeMounts:
+  - name: plugins-volume
+    mountPath: {{ include "turing.plugins.directory" . }}
+{{ end }}
+{{ end }}
+{{ end }}
+{{- end -}}
+
+{{- define "turing.initContainers" -}}
+initContainers:
+{{ with (include "turing.plugins.initContainers" . | fromYaml) -}}
+{{ if .initContainers }}
+{{- toYaml .initContainers -}}
+{{ end }}
+{{- end }}
+{{ with .Values.deployment.extraInitContainers }}
+{{- toYaml . -}}
+{{- end }}
+{{- end -}}
+
+{{- define "turing.environments" -}}
+{{ .Values.environmentConfigs | toYaml }}
+{{- end -}}
+
+{{- define "turing.environments.directory" -}}
+/app/cluster-env
+{{- end -}}
+
+{{- define "turing.environments.absolutePath" -}}
+{{- $base := include "turing.environments.directory" . -}}
+{{- $path := ternary .Values.mlp.environmentConfigSecret.envKey .Values.clusterConfig.environmentConfigPath (ne "" .Values.mlp.environmentConfigSecret.name) -}}
+{{- printf "%s/%s" $base $path -}}
+{{- end -}}
+
+{{/*
+Postgres related
+*/}}
+
+{{- define "turing-postgresql.host" -}}
+{{- if index .Values "turing-postgresql" "enabled" -}}
+  {{- printf "%s-turing-postgresql.%s.svc.cluster.local" .Release.Name .Release.Namespace -}}
+{{- else if .Values.turingExternalPostgresql.enabled -}}
+  {{- .Values.turingExternalPostgresql.address -}}
+{{- else -}}
+  {{- printf "%s-postgresql.%s.svc.cluster.local" .Release.Name .Release.Namespace -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "turing-postgresql.database" -}}
+  {{- if index .Values "turing-postgresql" "enabled" -}}
+    {{- index .Values "turing-postgresql" "postgresqlDatabase" -}}
+  {{- else if .Values.turingExternalPostgresql.enabled -}}
+    {{- .Values.turingExternalPostgresql.database -}}
+  {{- else -}}
+    {{- .Values.global.turing.postgresqlDatabase -}}
+  {{- end -}}
+{{- end -}}
+
+{{- define "turing-postgresql.username" -}}
+  {{- if index .Values "turing-postgresql" "enabled" -}}
+    {{- index .Values "turing-postgresql" "postgresqlUsername" -}}
+  {{- else if .Values.turingExternalPostgresql.enabled -}}
+    {{- .Values.turingExternalPostgresql.username -}}
+  {{- else -}}
+    {{- .Values.global.postgresqlUsername -}}
```

Comment:
Just curious, should this be `{{- .Values.global.turing.postgresqlUsername -}}` instead?

Reply:
No, it is actually meant to be a single set of "global" configs for the DB. It's set up the same way for Merlin.